src/client/serialization.ts

```diff
@@ -0,0 +1,104 @@
+export type MessageFormat = "none" | "double" | "float" | "integer" | "long" | "short";
+
+export type SerializationdResult = any | Error;
+
+class SerializationException extends Error { }
+
+// ---------------- Deserializers ----------------
+
+interface Deserializer {
+    deserialize(data: Buffer): any;
+}
+
+const deserializerRegistry: Map<MessageFormat, Deserializer> = new Map();
+
+export function deserialize(data: Buffer | null, format?: MessageFormat): SerializationdResult | null {
+    if (data === null || !format || format === "none") {
+        return null;
+    }
+    try {
+        const deserializer = getDeserializer(format);
+        if (!deserializer) {
+            throw new SerializationException(`Cannot find a deserializer for ${format} format.`);
+        }
+        return deserializer.deserialize(data);
+    }
+    catch (e) {
+        return e;
+    }
+}
+
+function getDeserializer(format: MessageFormat): Deserializer | undefined {
+    return deserializerRegistry.get(format);
+}
+
+class DoubleDeserializer implements Deserializer {
+
+    deserialize(data: Buffer | null): any {
+        if (data === null) {
+            return null;
+        }
+        if (data.length !== 8) {
+            throw new SerializationException("Size of data received by Deserializer is not 8");
+        }
+        return data.readDoubleBE(0);
+    }
+}
+
+class FloatDeserializer implements Deserializer {
+
+    deserialize(data: Buffer | null): any {
+        if (data === null) {
+            return null;
+        }
+        if (data.length !== 4) {
+            throw new SerializationException("Size of data received by Deserializer is not 4");

```

Comment:
...by FloatDeserializer...

Reply:
I could do that but it's the same message than Java FloatDeserializer.

Reply:
not a problem

Reply:
fixed